packages/engine/errors/8016.md

```diff
@@ -0,0 +1,6 @@
+---
+original: "Type assertion expressions can only be used in TypeScript files."
+excerpt: "This isn't a Typescript file"
```

Comment:
Add more detail here so it can act as a drop-in replacement for the original above.